### PR TITLE
finished lab w4

### PR DESCRIPTION
**Pull request by potatchipsxp**

### CODE REVIEW

File: lab.R [hunk 1]
```diff
@@ -16,13 +16,16 @@
...
 #' 
 
+#testing
+
 ## Setup
 ## **IMPORTANT**: Add all dependencies to `DESCRIPTION`
 library(tidyverse)
 library(skimr)
 library(visdat)
+library(dplyr)
```
Comment (dhicks) [inline lab.R:26]: Loaded as part of the tidyverse
```suggestion
```

File: lab.R [hunk 2]
```diff
@@ -49,23 +53,30 @@ dataf = read_csv(file.path('data', 'ames.csv'))
 #' 
 #' 1. The paper abstract (see above) reports 2930 rows.  How many observations (rows) are in our version of the dataset?  
 #' 
-problem2.1 = 1.7e15 # scientific notation: 1.7 x 10^15
+problem2.1 = 2930 # scientific notation: 1.7 x 10^15
+
+skimr::skim(dataf)
+#the skimr dunct gives a print out that says 293- is there a functional p[rogramming way of accessing that part of the returned tibble to set it as the answer?
```
Comment (dhicks) [inline lab.R:59]: There is, but I assumed you'd just type the values in, following the model

File: lab.R [hunk 3]
```diff
@@ -49,23 +53,30 @@ dataf = read_csv(file.path('data', 'ames.csv'))
...
+
 
 #' 2. The abstract also reports 80 "explanatory variables (23 nominal, 23 ordinal, 14 discrete, and 20 continuous)."  In R, factors are used for both nominal and ordinal; and numerics are used for both discrete and continuous. How many of each are in our version?
 #' 
-problem2.2.factors = 7
-problem2.2.characters = 18000
-problem2.2.numerics = 12
+#' 
+#' for the number of factors it says there are 81 columns, I am assuming one of those is the index column and thus we have 80 variables. its hard to say though looking at skimr output. 
+problem2.2.factors = 80
```
Comment (dhicks) [inline lab.R:67]: "Factor" is a specific type of variable in R, like "numeric" and "character" are types

File: lab.R [hunk 4]
```diff
@@ -36,11 +39,12 @@ dataf = read_csv(file.path('data', 'ames.csv'))
 #' 1. Look through the short descriptions in `?ames_raw` (or online, <https://cran.r-project.org/web/packages/AmesHousing/AmesHousing.pdf>).  Which variable reports the actual sale price? 
 #' 
 #' 
+#' it seems like value, except that it says that is a tibble so I think that isa tibble which i think means its another data table of value stuff? makes zero sense to me how this is organized
```
Comment (dhicks) [inline lab.R:42]: Try searching for "price," either in the docs or the codebook: <http://jse.amstat.org/v19n3/decock/DataDocumentation.txt>

File: lab.R [hunk 5]
```diff
@@ -114,23 +129,33 @@ distinct(dup_demo)
 #' Use this function to create a dataframe `dataf_nodup` with the duplicate rows removed.  Do `dataf` and `dataf_nodup` have the same number of rows?  What does this tell you about duplicate entries in our dataset?  
 #' 
 #' 
-#' 
-# dataf_nodup = ???
+#' they do. is there a way to do a tibble.shape?
+# dataf_nodup = 2930
```
Comment (dhicks) [inline lab.R:133]: `dim(dataf)` gives rows and columns; `nrow(dataf)` gives number of rows.  Uncomment this line for the automatic checker.

File: lab.R [hunk 6]
```diff
@@ -139,18 +164,23 @@ levels(bad_factor)
 
 #' To fix this, we need to set the order of the levels manually.  First, edit the next line, so that the levels correspond to the list in the documentation, with "Very Poor" as the lowest (first) value and "Very Excellent" as the highest (<http://jse.amstat.org/v19n3/decock/DataDocumentation.txt>, search for "Overall Cond").  
 
-condition_levels = c('Above_Average', 'Very_Poor', 'Below_Average')
+
+# "Above_Average" "Average"       "Below_Average" "Excellent"     "Fair"          "Good"          "Poor"          "Very_Good"     "Very_Poor"
+
+condition_levels = c("Very_Poor", "Poor", "Fair", "Below_Average", "Average", "Above_Average", "Good", "Very_Good", "Excellent")
```
Comment (dhicks) [inline lab.R:170]: You're missing "Very_Excellent" (in the codebook but not the dataset)

File: lab.R [hunk 7]
```diff
@@ -139,18 +164,23 @@ levels(bad_factor)
...
+good_factor = factor(dataf$Overall_Cond, levels = condition_levels)
 
 #' 5. Finally we want this factor to be in our analysis dataframe.  *Normally, to preserve immutability*, I would do this right after reading in the CSV (using a pipe).  We'll talk more about this when we work on cleaning data.  For this lab, just do it here.  
 #' 
 #' Using `mutate()`, add a variable `overall_cond_fct` to `dataf`, that represents overall condition as a factor with the levels in the correct order. 
 
 # dataf = ???
 
+dataf %>% 
+    mutate(overall_cond_fct = good_factor)
```
Comment (dhicks) [inline lab.R:183]: You need to assign the result to a variable (in this case `dataf`)
```suggestion
dataf = dataf %>% 
    mutate(overall_cond_fct = good_factor)
```

File: lab.R [hunk 8]
```diff
@@ -177,18 +209,22 @@ select(dataf, Sale_Price, Overall_Cond, Gr_Liv_Area)
...
+cor_matrix = dataf_smol %>%
+    mutate_if(is.factor, as.numeric) %>%
+    cor(method = 'spearman')
+
 #' 3. Now we convert the correlation matrix into a dataframe.  
 #' a. Explain what the following line of code is doing.  Hint: Read the docs! 
 #' 
 #' 
-#' 
+#' it read the matreix as a tibble. like aslist or asarray. Is there something more menaingful im missing here?
```
Comment (dhicks) [inline lab.R:220]: What does the `rownames` argument do?

File: lab.R (hunk 7 above)
Comment (dhicks) [reply lab.R:183]: It looks like most of the errors further down happen because you didn't assign this back to `dataf`.

File: lab.R (hunk 1 above)
Comment (potatchipsxp) [reply lab.R:26]: It gave me an error when I didnt include it

File: lab.R (hunk 4 above)
Comment (potatchipsxp) [reply lab.R:42]: I misunderstood this question, sorry

File: lab.R (hunk 6 above)
Comment (potatchipsxp) [reply lab.R:170]: oops sorry.  I just printed our the ones in the dataset and assumed that was all of them

**Pull request closed without merging**